File: src/leetcode/2026/05/Solution944.java
public class Solution944 {
    public int minDeletionSize(String[] strs) {
        int count = 0;
        int n = strs[0].length(); // length of each string
        int m = strs.length; // number of strings
        for (int i = 0; i < n; i++) {
            for (int j = 1; j < m; j++) {
                if (strs[j].charAt(i) < strs[j - 1].charAt(i)) {
                    count++;
                    break;
                }
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Solution944 solution = new Solution944();

        String[] strs1 = { "cba", "daf", "ghi" };
        int result1 = solution.minDeletionSize(strs1);
        System.out.println("Output for strs1: " + result1); // Output: 1

        String[] strs2 = { "a", "b" };
        int result2 = solution.minDeletionSize(strs2);
        System.out.println("Output for strs2: " + result2); // Output: 0

        String[] strs3 = { "zyx", "wvu", "tsr" };
        int result3 = solution.minDeletionSize(strs3);
        System.out.println("Output for strs3: " + result3); // Output: 3
    }
}

File: src/leetcode/2026/05/Solution944.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution944 {
public:
    int minDeletionSize(vector<string>& strs) {
        int count = 0;
        int n = strs.size(); //number of strings
        int m = strs[0].size(); //length of each string

        for (int j = 0; j < m; j++) {
            for (int i = 1; i < n; i++) {
                if (strs[i][j] < strs[i - 1][j]) {
                    count++;
                    break;
                }
            }
        }

        return count;
    }
};

int main() {
    Solution944 solution;

    vector<string> strs1 = {"cba", "daf", "ghi"};
    vector<string> strs2 = {"a", "b"};
    vector<string> strs3 = {"zyx", "wvu", "tsr"};

    cout << solution.minDeletionSize(strs1) << endl; // Output: 1
    cout << solution.minDeletionSize(strs2) << endl; // Output: 0
    cout << solution.minDeletionSize(strs3) << endl; // Output: 3

    return 0;
}